pkgs/tools/security/gitsign/default.nix

```diff
@@ -1,19 +1,28 @@
-{ lib, buildGoModule, fetchFromGitHub, stdenv }:
+{ lib, buildGoModule, fetchFromGitHub, stdenv, gitMinimal, makeWrapper }:
 
 buildGoModule rec {
   pname = "gitsign";
-  version = "0.3.1";
+  version = "0.3.2";
 
   src = fetchFromGitHub {
     owner = "sigstore";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-jL39U7kmdy5lo/caetpFlFCLfby6ftMWDMPq6yt3nUk=";
+    sha256 = "sha256-hDVn7ZiZoY0FSgIsApZliMIq1xjuNdg+DMvKzP5kET0=";
   };
-  vendorSha256 = "sha256-9tmENlR3vVELmT7JWcz7RhtIji7go2V/Bsu9nMrvHfs=";
+  vendorSha256 = "sha256-5hVcul5DlHZ0Gtw1LdBmxGpsmuD2bTtwPGysOUwe2k0=";
+
+  buildInputs = [ gitMinimal ];
+  nativeBuildInputs = [ makeWrapper ];
 
   ldflags = [ "-s" "-w" "-buildid=" "-X github.com/sigstore/gitsign/pkg/version.gitVersion=${version}" ];
 
+  postInstall = ''
+    wrapProgram $out/bin/gitsign --prefix PATH : ${lib.makeBinPath [ gitMinimal ]}
```

Comment:
git should be used from its environment like it is expected by Nix.

Reply:
Sorry, what does this mean exactly?

Reply:
So you recommend to not wrap gitsign with Git?

I find it a bit strange since in this situation it is a dependency like any other. They were using `go-git` until now but have been forced to call the git binary in order to mitigate some issues.

Reply:
Yes, I would avoid wrapping Git., but it is not wrong to wrap.

My ideas, why I would not wrap:
1. Git is only a runtime dependency, not a build time. Rebuild of Git leads to a rebuild of package gitsign.
2. Git and Nvim are interactive user tools not libraries. When a user want to use Git, the user has to install it anyway.
3. When we wrap scripts in gitsign, did we found all parts to wrap? If we missed one, then it is hard to find a possible issue.

Reply:
> Git is only a runtime dependency, not a build time. Rebuild of Git leads to a rebuild of package gitsign.

I'm not really sure to see how this is an issue? The docs seems to say `buildInputs` is the appropriate for dependencies used at run-time

https://github.com/NixOS/nixpkgs/blob/83b198a2083774844962c854f811538323f9f7b1/doc/stdenv/stdenv.chapter.md?plain=1#L220

Existing tools around Git also uses `makeWrapper`, see for example [`git-secret`](https://github.com/NixOS/nixpkgs/blob/83b198a2083774844962c854f811538323f9f7b1/pkgs/applications/version-management/git-and-tools/git-secret/default.nix#L20), [ `git-octopus`](https://github.com/NixOS/nixpkgs/blob/83b198a2083774844962c854f811538323f9f7b1/pkgs/applications/version-management/git-and-tools/git-octopus/default.nix#L16) or [`git-crypt`](https://github.com/NixOS/nixpkgs/blob/83b198a2083774844962c854f811538323f9f7b1/pkgs/applications/version-management/git-and-tools/git-crypt/default.nix#L47). This approach does not seem that uncommon.

> Git and Nvim are interactive user tools not libraries. When a user want to use Git, the user has to install it anyway.

Git also has "plumbing" commands that are stable and can be used in scripts.

If we do not have Git in our runtime dependencies it also means that doing thing like checking that calling `gitsign` will end with an error:
```
$ nix-shell --pure -p gitsign --run 'gitsign --version'
error reading config: exec: "git": executable file not found in $PATH:
```
I'm not sure this something we want.

For the `gitsign` binary itself yes it is unlikely we end up in a situation where we do not have `git` in on the PATH. For things like [`gitsign-attest`](https://github.com/sigstore/gitsign/tree/main/cmd/gitsign-attest#usage) it is less obvious: I could see myself generating the SBOM of one of my app and call `gitsign-attest` in a script that does not have `git`.

> When we wrap scripts in gitsign, did we found all parts to wrap? If we missed one, then it is hard to find a possible issue.

Well, I have been able to sign my commit with it so at least the base feature is working.
It is also the only usage of `os/exec` they do in their codebase so yes I am fairly confident we found everything.

Also I am not sure to see how this is specific to `gitsign`, do not we have the same potential issues for anything calling another executable?
For Go code this could probably be improved by doing some static analysis but this is likely to consume quite a lot of resources and I would not be surprised if it generated too much false alarms to be useful.

Reply:
We can find examples for both sides in nixpkgs. Following example packages avoid wrapping the cli tools:
- [git-team](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/version-management/git-and-tools/git-team/default.nix)
- [git-workspace](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/version-management/git-and-tools/git-workspace/default.nix)

I think both ways are valid.